lib/protocol_9p_server.ml

```diff
@@ -246,6 +246,7 @@ module Make(Log: Protocol_9p_s.LOG)(FLOW: V1_LWT.FLOW)(Filesystem: Protocol_9p_f
     LowLevel.expect_version ~write_lock reader writer
     >>*= fun (tag, v) ->
     let msize = min msize v.Request.Version.msize in
+    let msize = min msize Protocol_9p_buffered9PReader.max_message_size in
```

Comment:
Should this be `min` of all three values? I think the requested `msize` should be taken into account when deciding the `msize` of the connection.